control/controlhttp/client.go

```diff
@@ -402,6 +305,9 @@ func (a *Dialer) dialHostOpt(ctx context.Context, optAddr netip.Addr, optACEHost
 	}
 
 	var err80, err443 error
+	if forceTLS {
+		err80 = errors.New("TLS forced: no port 80 dialed")
+	}
```

Comment:
I liked my version of this without the allocation and misleading "all connection attempts failed" when there was only one singular attempt

Reply:
u80 wasn't nil though, so it wasn't actually working, but I agree we could refactor this more

Reply:
ah, it's u443 that gets niled when not used. yeah, that's inconsistent.